From b39f5c1ddc4ddaf04dbc95e729b760def3523e48 Mon Sep 17 00:00:00 2001
From: Eidolon <furyhunter600@gmail.com>
Date: Sun, 12 Jun 2022 00:01:16 -0500
Subject: [PATCH] Fix buffer overflow in chat insert-paste

---
 src/hu_stuff.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/hu_stuff.c b/src/hu_stuff.c
index 5d893a551..9b973c93c 100644
--- a/src/hu_stuff.c
+++ b/src/hu_stuff.c
@@ -1089,6 +1089,7 @@ boolean HU_Responder(event_t *ev)
 			const char *paste;
 			size_t chatlen;
 			size_t pastelen;
+			size_t movelen;
 
 			if (CHAT_MUTE)
 				return true;
@@ -1102,7 +1103,10 @@ boolean HU_Responder(event_t *ev)
 			if (chatlen+pastelen > HU_MAXMSGLEN)
 				return true; // we can't paste this!!
 
-			memmove(&w_chat[c_input + pastelen], &w_chat[c_input], pastelen);
+			// prevent moving past the chat buffer by truncating
+			movelen = min(chatlen + pastelen, HU_MAXMSGLEN) - c_input;
+			memmove(&w_chat[c_input + pastelen], &w_chat[c_input], movelen);
+
 			memcpy(&w_chat[c_input], paste, pastelen); // copy all of that.
 			c_input += pastelen;
 			return true;
-- 
GitLab

