From 5668c838053e533089f38ea00308c675450872bb Mon Sep 17 00:00:00 2001
From: Shane Ellis <cobaltbw@gmail.com>
Date: Thu, 8 Jul 2021 18:07:34 -0400
Subject: [PATCH] Add lives check to Team name HUD

---
 src/st_stuff.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/st_stuff.c b/src/st_stuff.c
index af14118e3..dff7c562a 100644
--- a/src/st_stuff.c
+++ b/src/st_stuff.c
@@ -857,7 +857,7 @@ static void ST_drawLivesArea(void)
 		}
 	}
 	// Team name
-	else if (G_GametypeHasTeams())
+	else if (G_GametypeHasTeams() && !(gametyperules & GTR_LIVES))
 	{
 		if (stplyr->ctfteam == 1)
 		{
-- 
GitLab

